README.md

```diff
@@ -51,5 +51,5 @@ reasoning, to this markdown file.
 - Worst Case Memory Complexity : $O(n * 2^n)$
 * the cache stores $2^n$ keys every call, therefore the cache will store $n * 2^n$ elements when the algorithm is done running since we may have multiple of the same permutations that are unique because they will have different start values.  
 - Worst Case Time Complexity : $O(n * 2^n)$
-* since we reduce the number of subproblems stored in the cache by using memoization, the number of calls here would not be n! but instead would be 2^n. And since we need to check all start options, in the worst case, out time complexity will end up being $O(n * 2^n)$
+* For each cache entry we create a 'newcities' which takes time complexity of $O(n)$. We also run a 'for' loop 'n' times to find the next starting point, and the new 'newcities' array. Therefore for each cache entry we must have time complexity of $O(n + n)$. It is worth noting that our for loop inside the tsp function will not necessarly run 'n' times where n is the size of the input matrix, but where 'n' is the size of the 'nextcites' array, which will always be less than or equal to the size of the input array. Since we reduce the number of subproblems stored in the cache by using memoization, the number of calls here would not be n! but instead would be 2^n. And since we need to check all start options, in the worst case, our time complexity will end up being $O((n + n) * (n * 2^n))$ which can be simplified to $O(n * 2^n)$. 
```

Comment:
Not following you here. Just above you say that the memory complexity is the same, i.e. there are that many cache entries?